src/ring_theory/graded_algebra/basic.lean

```diff
@@ -103,6 +103,37 @@ dfinsupp.mem_support_iff.trans add_submonoid_class.coe_eq_zero.not.symm
 
 end graded_ring
 
+section add_left_cancel_monoid
+
+open direct_sum
+
+lemma direct_sum.decompose_mul_add_of_left_mem {ι σ A}
+  [decidable_eq ι] [add_left_cancel_monoid ι] [semiring A]
+  [set_like σ A] [add_submonoid_class σ A] (𝒜 : ι → σ) [graded_ring 𝒜]
+  {i : ι} (a : 𝒜 i) {b : A} {j : ι} :
+  decompose 𝒜 (↑a * b) (i + j) =
+    @graded_monoid.ghas_mul.mul ι (λ i, 𝒜 i) _ _ _ _ a (decompose 𝒜 b j) :=
+begin
+  ext,
```

Comment:
If you're going to prove this via ext (maybe that is easiest), you may as well prove the coerced version first, and then prove this version via `subtype.ext $ that_version _ _`